service/history/shard/context_factory.go

```diff
@@ -116,6 +119,11 @@ func (c *contextFactoryImpl) CreateContext(
 		c.HostInfoProvider,
 		c.TaskCategoryRegistry,
 	)
+	if shard.GetConfig().EnableHostLevelEventsCache() {
+		shard.eventsCache = c.EventsCache
+	} else {
+		shard.eventsCache = c.NewEventsCacheFn(shard.executionManager, shard.config, shard.metricsHandler, shard.contextTaggedLogger)
+	}
```

Comment:
I think the logic should be inside `newContext()` and `c.EventsCache` can be an argument?

Reply:
RIght. Made the change.